projects/organization-management/src/app/store/users/users.actions.ts

```diff
@@ -23,4 +32,61 @@ export class LoadUsersSuccess implements Action {
   constructor(public payload: { users: User[] }) {}
 }
 
-export type UsersAction = LoadUsers | LoadUsersFail | LoadUsersSuccess;
+export class AddUser implements Action {
+  readonly type = UsersActionTypes.AddUser;
+  constructor(public payload: { user: User; successMessage?: string; successRouterLink?: string }) {}
+}
+
+export class AddUserFail implements Action {
+  readonly type = UsersActionTypes.AddUserFail;
+  constructor(public payload: { error: HttpError }) {}
+}
+
+export class AddUserSuccess implements Action {
+  readonly type = UsersActionTypes.AddUserSuccess;
+  constructor(public payload: { newUser: User; successMessage?: string }) {}
+}
+
+export class UpdateUser implements Action {
+  readonly type = UsersActionTypes.UpdateUser;
+  constructor(public payload: { user: User; successMessage?: string; successRouterLink?: string }) {}
+}
+
+export class UpdateUserFail implements Action {
+  readonly type = UsersActionTypes.UpdateUserFail;
+  constructor(public payload: { error: HttpError }) {}
+}
+
+export class UpdateUserSuccess implements Action {
+  readonly type = UsersActionTypes.UpdateUserSuccess;
+  constructor(public payload: { user: User; successMessage?: string }) {}
```

Comment:
successMessage is not necessary if you trigger the SuccessMessage action in the UpdateUserSuccess Effect effect